File: backend/src/timeflow/infrastructure/websocket/messages/reminder.py
"""提醒下发相关消息。"""

from typing import Literal

from pydantic import BaseModel

from timeflow.infrastructure.websocket.messages.envelope import ErrorDetail


class ReminderControl(BaseModel):
    """服务端主动下发:该提醒了,具体展示方式由客户端决定。"""

    type: Literal["reminder.control"] = "reminder.control"
    schedule_id: str
    reason: str
    action: str


class ReminderControlAck(BaseModel):
    """客户端执行提醒后回传的确认。"""

    type: Literal["reminder.control.ack"] = "reminder.control.ack"
    schedule_id: str
    ok: bool
    error: ErrorDetail | None = None


class ReminderAudioStart(BaseModel):
    """服务端开始下发一个提醒音频流。"""

    type: Literal["reminder.audio.start"] = "reminder.audio.start"
    schedule_id: str
    stream_id: str
    audio_format: str


class ReminderAudioEnd(BaseModel):
    """服务端结束一个提醒音频流。"""

    type: Literal["reminder.audio.end"] = "reminder.audio.end"
    schedule_id: str
    stream_id: str


class ReminderAudioAck(BaseModel):
    """客户端回传提醒音频接收或播放结果。"""

    type: Literal["reminder.audio.ack"] = "reminder.audio.ack"
    schedule_id: str
    stream_id: str
    ok: bool
    error: ErrorDetail | None = None


__all__ = [
    "ReminderAudioAck",
    "ReminderAudioEnd",
    "ReminderAudioStart",
    "ReminderControl",
    "ReminderControlAck",
]

File: frontend/src/contracts/reminder.ts
import type { ApiError } from './envelope';

/**
 * 提醒通道协议（预留）。
 * 当前前端尚未接入 WS 提醒控制 / TTS 音频流；类型仅作与后端对齐的权威契约文档。
 * 接入客户端前请勿在业务层依赖这些消息。
 */

/** 服务端下发提醒控制；具体展示通道由客户端根据前后台自行决定。 */
export type ReminderControl = {
  type: 'reminder.control';
  schedule_id: string;
  reason: string;
  action: 'show';
};

export type ReminderControlAck =
  | { type: 'reminder.control.ack'; schedule_id: string; ok: true }
  | { type: 'reminder.control.ack'; schedule_id: string; ok: false; error: ApiError };

/** 提醒 TTS 音频流开始；随后通过同一 WebSocket 连接发送 Binary Frame。 */
export type ReminderAudioStart = {
  type: 'reminder.audio.start';
  schedule_id: string;
  stream_id: string;
  audio_format: 'mp3';
};

export type ReminderAudioEnd = {
  type: 'reminder.audio.end';
  schedule_id: string;
  stream_id: string;
};

export type ReminderAudioAck =
  | {
      type: 'reminder.audio.ack';
      schedule_id: string;
      stream_id: string;
      ok: true;
    }
  | {
      type: 'reminder.audio.ack';
      schedule_id: string;
      stream_id: string;
      ok: false;
      error: ApiError;
    };
